PlatformRequirements.md

```diff
@@ -0,0 +1,144 @@
+    - options : This is an options object. The most common options are :
+        - options.customTemplate : This is a path to custom template, that should override the default one from platform.
+        - options.link : This is a flag that should indicate that the platform's sources will be linked to the installed platform instead of copying.
+     - The `updatePlatform` method must return a promise, which is either fulfilled with a PlatformApi instance or rejected with a CordovaError.
+ 
+- Api.js exports static function `createPlatform(destination, cfg, options, events);`
+    - PlatformApi.createPlatform = function(cordovaProject, options) {};
+    - The `createPlatform method` is equal to the bin/create script. It should install the platform to a specified directory and create a platform project. It should accept a CordovaProject instance, that defines a project structure and configuration, that should be applied to the new platform, and an options object.
+    - cordovaProject : This is a CordovaProject instance that defines a project structure and configuration, that should be applied to the new platform. This argument is optional and if not defined, that platform is used as a standalone project and not as part of a Cordova project.
+    - options : This is an options object. The most common options are :
+        - options.customTemplate : This is a path to custom template, that should override the default one from the platform.
+        - options.link : This is a flag that should indicate that the platform's sources will be linked to the installed platform instead of copying.
+    - The `createPlatform` method must return a promise, which is either fulfilled with a PlatformApi instance or rejected with a CordovaError.
+ 
+The way most platforms work is somewhat tricky.  The Api.js could be anywhere in the platform repo, ex. /templates/cordova/Api.js .  When a new project is created for the platform, the platform copies this file (and supporting files ) to destination/cordova/Api.js.  The project expectations demand that the Api.js file be available at /projectRoot/platforms/platform-name/cordova/Api.js.
+ 
+A call to the platforms static Api.createPlatform will
+1. Copy template files to destination (according to cfg, options).
+1. Copy the file Api.js (and supporting modules) to destination/cordova/Api.js.
+1. Return a promise that will eventually resolve with an instance of the newly copied Api.js.
+    - This step is a bit confusing because at this point in time we have 2 Api.js in memory, one from the platform folder which has static methods, and one from the new project folder which has instance methods.
+    - Similarly, they may have their own require() statements, and while typically require would return the same instance, because our modules are loaded from different paths, they get different relative requires also.
+ 
+ 
+## Api (Project) Expectations
+ 
+- Api.js exports a constructor function
```

Comment:
This is NOT a requirement.  All instances of the platform api are typically created and returned by the createPlatform function.

The newly created platform ( which copies code from the template to the project dir ) MUST export a constructor though.

Reply:
Do we still want to add that info tidbit to the document maybe?

Reply:
I don't think its that important, just merge

Reply:
It had also be added below that headline.